src/lib/plugin.js

```diff
@@ -525,16 +525,14 @@ module.exports = class PluginPaymentChannel extends EventEmitter2 {
     await this._transfers.cancel(transferId, reason)
     this.debug('rejected ' + transferId)
 
-    // const rejectionReason = ilpPacket.serializeIlpError({
-    //   code: 'F00', // TODO: what should be the code? (cc: sharafian)
-    //   name: 'Bad Request', // TODO: what should be the name?   (cc: sharafian)
-    //   triggeredBy: this.getAccount(),
-    //   forwardedBy: [],
-    //   triggeredAt: new Date(),
-    //   data: reason
-    // })
-
-    const rejectionReason = ilpPacket.serializeIlpError(reason)
+    const rejectionReason = ilpPacket.serializeIlpError({
+      code: reason.code,
+      name: reason.name,
+      triggeredBy: reason.triggered_by,
+      forwardedBy: reason.forwarded_by,
+      triggeredAt: reason.triggered_at,
+      data: reason.additional_info
```

Comment:
I think we need a JSON.stringify here... LPI says this is an object, while ilp-packet expects an ASCII string.

And I think that may be an issue with ilp-packet as well, since JSON can by any UTF-8, so it should probably be UTF-8, not ASCII. But even that is weird, because we may need errors that have machine-readable data... (e.g. #318)

For now, I'd add a JSON.stringify here, that'll make most things mostly work. Then we can figure out the proper fix.

Reply:
:+1 done